### ISSUE TITLE
Add Facility Name, Testing Lab Name, and Testing Lab CLIA to Azure Logging observability

### ISSUE THREAD

**Issue by chris-kuryak** [CDCgov/prime-reportstream#16400]
### User Story
As a ReportStream Engineer,
I want to be able to query for more information (Facility Name, Testing Lab Name, Testing Lab CLIA) at the item level in Azure logs,
so that I can troubleshoot issues easier.

### Description/Use Case
Sometimes we receive reports that include multiple testing facilities (such as Carbon Health) and having the information of Facility Name, Testing Lab Name, and Testing Lab CLIA will allow us to identify specific messages easier and troubleshoot issues faster.

### Dev Notes
Currently, we track items at the Report level and Item level. The added observability with these fields we would like to be added at the item level.

Here is an example in `FHIRDestinationFilter.kt` where we are sending an Item event, specifically `ITEM_ROUTED`:

```kotlin
    reportEventService.sendItemEvent(
        eventName = ReportStreamEventName.ITEM_ROUTED,
        childReport = report,
        pipelineStepName = TaskAction.destination_filter
    ) {
        parentReportId(queueMessage.reportId)
        params(
            mapOf(
            ReportStreamEventProperties.RECEIVER_NAME to receiver.fullName,
            ReportStreamEventProperties.BUNDLE_DIGEST
                to bundleDigestExtractor.generateDigest(bundle)
        )
        )
        trackingId(bundle)
    }
```

As you can see, `BUNDLE_DIGEST` is passed to the `params` section of the event data, and the object returned by `generateDigest` looks like so:

```kotlin
data class BundleDigestLabResult(
    val observationSummaries: List<ObservationSummary>,
    val patientState: List<String>,
    val performerState: List<String>,
    val orderingFacilityState: List<String>,
    override val eventType: String,
) : BundleDigest
```

Therefore, this work probably revolves around expanding this data class and the factory that creates it: `FhirPathBundleDigestLabResultExtractorStrategy`

I confirmed with Victor that performer/testing lab and order facility data should be correlated. So that means we need a more complicated object to store the info than we use today, like so:

```kotlin
[
{
    performerName: "something"
    performerState: "NA"
    performerCLIA: "Dfds1234"
},
{
    performerName: "something2"
    performerState: "NA"
    performerCLIA: "a234o23"
}
]
```

```kotlin
[
{
    orderingFacilityName: ""
    orderingFacilityState: ""
},
{
    orderingFacilityName: ""
    orderingFacilityState: ""
}
]
```

### Acceptance Criteria
- [ ] A query in Azure Logs for any `Item` events includes the Facility Name, Testing Lab Name, and Testing Lab CLIA and those fields that can also be used in the query to target specific items.
- [ ] Azure Event documentation in LucidChart is updated accordingly: https://lucidgov.app/lucidchart/89edc4c3-695b-4324-be8e-6556dd9fec95/edit?invitationId=inv_a44a9145-fc99-49ca-9053-a873937c7b23
- [ ] An example query for "List of items from Facility Name 'X'" is added to the bottom of the Azure Events document in Github: https://github.com/CDCgov/prime-reportstream/blob/master/prime-router/docs/observability/azure-events.md

**Comment by chris-kuryak** [comment]
@MichaelEsuruoso @brandonnava @arnejduranovic 

Made this ticket for adding some additional fields to be tracked in Azure Logging.

Does this meet y'alls needs for enough information? Does it have the right labels, column, etc? Let me know if there is anything missing or if there is a better avenue for these requests so I can make sure I get it right next time.

**Comment by MichaelEsuruoso** [comment]
information looks. for the AC just to make sure I understand, do you want us to add a search query for the items listed?

**Comment by chris-kuryak** [comment]
@MichaelEsuruoso Good question and feedback.

I have added specific ACs around documentation regarding this update. Let me know if that provides enough guidance.

**Comment by MichaelEsuruoso** [comment]
this look good thank you @chris-kuryak  not sure if @arnejduranovic or  @brandonnava  have anything to add. Also what is the priority for this?

**Comment by arnejduranovic** [comment]
@chris-kuryak presently we should be capturing the following in the item events:

```
    val performerState: List<String>,
    val orderingFacilityState: List<String>,
```

Do you want this data somehow associated with the new items you are requesting:

For example,

```
[
{
    facilityName: "something"
    facilityState: "NA"
},
{
    facilityName: "something2"
    facilityState: "NA"
}
]
```

and
 
```
[
{
    performerName: "something"
    performerState: "NA"
    performerCLIA: "Dfds1234"
},
{
    performerName: "something2"
    performerState: "NA"
    performerCLIA: "a234o23"
}
]
```
or does this kind of association not matter for your use case(s)? Keep in mind we do have a char limit of like 8000 for each param of the event and I think we would need to dig in and make sure we wouldn't run up against that limit if we did it this way.

**Comment by chris-kuryak** [comment]
@MichaelEsuruoso The priority is low right now. But as we ramp up more of Carbon Health's production data, this would become a medium priority. So if it's not implemented by the end of December it may become Medium priority if Carbon Health's volume is high.

@victor-chaparro Do you have any insight that could help guide @arnejduranovic per his comment above? I'm not sure what the advantages/disadvantages of the different methods as it affects our ability to query.

**Comment by chris-kuryak** [comment]
@arnejduranovic Yes, the above proposal looks good. @victor-chaparro gave me his blessing.

**Comment by arnejduranovic** [comment]
Apologies, Chris, but I am still not clear (perhaps that is my fault for the way I phrased the question). I would like to know from your team if it is okay for us to store the additional requested elements on their own, so like:

```
val performerState: List<String>,
val orderingFacilityState: List<String>,
val performerCLIA: List<String>
```

or do you have to have them associated like in the JSON block in my previous comment?

**Comment by MichaelEsuruoso** [comment]
we didnt get a. response for this ticket before refinement, so item was not refined.

**Comment by chris-kuryak** [comment]
@arnejduranovic Yes, if @victor-chaparro  and I are interpreting your question correctly, there may be multiple facilities in one item. Therefore, we believe that it needs to be included in JSON in order to maintain association at the Item/Observation level.

**Comment by jack-h-wang** [comment]
@MichaelEsuruoso issue spillover due to planned and unplanned absences this sprint.

**Comment by MichaelEsuruoso** [comment]
Potential blocker- issues with Kant testing in Azure due to him not having CDC network access.
